proposals/testing/NNNN-attachments.md

```diff
@@ -0,0 +1,436 @@
+# Attachments
+
+* Proposal: [ST-NNNN](NNNN-attachments.md)
+* Authors: [Jonathan Grynspan](https://github.com/grynspan)
+* Status: **Awaiting review**
+* Bug: [swiftlang/swift-testing#714](https://github.com/swiftlang/swift-testing/issues/714)
+* Implementation: [swiftlang/swift-testing#973](https://github.com/swiftlang/swift-testing/pull/973)
+* Review: ([pitch](https://forums.swift.org/t/pitch-attachments/78072))
+
+## Introduction
+
+Test authors frequently need to include out-of-band data with tests that can be
+used to diagnose issues when a test fails. This proposal introduces a new API
+called "attachments" (analogous to the same-named feature in XCTest) as well as
+the infrastructure necessary to create new attachments and handle them in tools
+like VS Code.
+
+## Motivation
+
+When a test fails, especially in a remote environment like CI, it can often be
+difficult to determine what exactly has gone wrong. Data that was produced
+during the test can be useful, but there is currently no mechanism in Swift
+Testing to output arbitrary data other than via `stdout`/`stderr` or via an
+artificially-generated issue. A dedicated interface for attaching arbitrary
+information to a test would allow test authors to gather relevant information
+from a test in a structured way.
+
+## Proposed solution
+
+We propose introducing a new type to Swift Testing, `Attachment`, that represents
+some arbitrary "attachment" to associate with a test. Along with `Attachment`,
+we will introduce a new protocol, `Attachable`, to which types can conform to
+indicate they can be attached to a test.
+
+Default conformances to `Attachable` will be provided for standard library types
+that can reasonably be attached. We will also introduce a **cross-import overlay**
```

Comment:
Consider linking to the draft proposal about cross-import overlays. I don't think it ever advanced beyond that stage, but it could still provide helpful context

https://forums.swift.org/t/cross-import-overlays/36710